recipes-support/provisioning/files/phytec-board-config.sh

```diff
@@ -326,6 +329,11 @@ The next steps are:
             set_onboarded accountdevice
         fi
         echo "response: ${response}"
+        echo "After the successful onboarding the awsclient needs to be restarted 3 times."
+        awsrestart="1"
+    fi
+    if [ "$awsrestart" = "0" ]; then
+        do_restart_awsclient
     fi
```

Comment:
Ok, I just remembered our discussion about this.

The problem here still is though, that the restart occurs immediately after a token is being requested, which is not ideal. The awsclient should be restarted after the phytec-board-config tool finishes to give the user enough time for onboarding their device.

Reply:
in the interactive mode the restart only occurs when the user presses "ok" after the token was generated it also says so in the tool. The if statement is needed, because otherwise when the user onboards via commandline the awsclient would restart immediately after the token is generated and as you said that would not make sense. I wrote a prompt so that would not happen but we decided that its not good to have the user interact when using commandline option. So now the awsclient will not be restarted after onboarding via commandline, merely a note will be displayed that the awsclient needs to be restarted.

Reply:
Ok, but you can still remove that if statement and move the restarting of the service to the interactive part. And there is also no need to create the function `do_restart_awsclient` as it is only used once.